app/models/embed/purl.rb

```diff
@@ -52,6 +52,10 @@ def downloadable_files
       all_resource_files.select(&:downloadable?)
     end
 
+    def transcript_files
```

Comment:
How about renaming to `downloadable_transcript_files?` and then use `any?` instead of `select`